### PR TITLE
backport: bitcoin#17934, #21338, #21390, #21445, #21602, #21606, #21609, #21676, bitcoin-core/gui#260, 

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
Regular backports from bitcoin v22

## Note for reviewers:
PRs bitcoin#17934 and bitcoin#21606 have been backported partially in past.

## What was done?
Removed unused sanitizer rules (see bitcoin#21366 and dashpay/dash#5055)
 - bitcoin/bitcoin#21338
 - bitcoin/bitcoin#21390
 - bitcoin/bitcoin#21609
 - bitcoin/bitcoin#21445
 - bitcoin/bitcoin#21606
 - bitcoin/bitcoin#17934
 - bitcoin-core/gui#260
 - bitcoin/bitcoin#21602
 - bitcoin/bitcoin#21676

## How Has This Been Tested?
Run unit/functional tests

## Breaking Changes
N/A

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by knst** [comment]
> @[knst](https://github.com/knst) knst [force-pushed](https://github.com/dashpay/dash/compare/f94a58d9afad6956bc5d86a601873fa229888373..aca303f74c5cb22b94b33fc6a9dc2f2b0d5130dc) the bp-v22-p6 branch from[ f94a58d ](https://github.com/dashpay/dash/commit/f94a58d9afad6956bc5d86a601873fa229888373)to aca303f
Rebased on the top of develop

CI before rebase succeeded: https://gitlab.com/dashpay/dash/-/pipelines/1245926027

File: test/lint/extended-lint-cppcheck.sh [hunk 1]
```diff
@@ -55,7 +55,6 @@ IGNORED_WARNINGS=(
     "src/test/checkqueue_tests.cpp:.* Struct 'UniqueCheck' has a constructor with 1 argument that is not explicit."
     "src/test/fuzz/util.h:.* Class 'FuzzedFileProvider' has a constructor with 1 argument that is not explicit."
     "src/test/fuzz/util.h:.* Class 'FuzzedAutoFileProvider' has a constructor with 1 argument that is not explicit."
-    "src/util/ref.h:.* Class 'Ref' has a constructor with 1 argument that is not explicit."
```
Comment (UdjinM6) [inline test/lint/extended-lint-cppcheck.sh:58]: 2e3b6404235cf753bd130fb58a4cb0022c06b8a0 : duplicate https://github.com/dashpay/dash/pull/5974/commits/5d8abea6bdb5734641836c207f072e6a171b7b73 in #5974

**Review by UdjinM6 — DISMISSED** [review]
one potential conflict, LGTM otherwise

utACK

File: test/lint/extended-lint-cppcheck.sh (hunk 1 above)
Comment (knst) [reply test/lint/extended-lint-cppcheck.sh:58]: removed from 5974

**Review by UdjinM6 — APPROVED** [review]
re-utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK 21ad71c5788017a9aad90205ebb4f79811d3f9ac

**Pull request merged**